### PR TITLE
iso_checksum: fix parsing with absent newline

### LINKED ISSUE #3383: iso_checksum_url does not allow md5 sum without newline at the end
«iso_checksum_url (string) - A URL to a GNU or BSD style checksum file containing a checksum for the OS ISO file»

When «iso_checksum_url» contains content with newline at the end, all OK
«0fb3f1fc9bfd1a423518da6305a79f73  some-iso-image.iso\n»

When no newline at the end, just 
«0fb3f1fc9bfd1a423518da6305a79f73  some-iso-image.iso»

packer failed with 
«No checksum for "some-iso-image.iso" found at: … »

- Packer Version: 0.9.0
- Host platform: Linux/Windows/ALL
- Debug log output from `PACKER_LOG=1 packer build template.json`.
  https://gist.github.com/belonesox/92b25b18bca9ae9c53a7
- The _simplest example template and scripts_ needed to reproduce the bug.
   https://gist.github.com/belonesox/bc47abb20bb50b9085f7

### PR DESCRIPTION
**Pull request by vtolstov**
Fix iso checksum parsing when file does not have newline

Closes #3383

Signed-off-by: Vasiliy Tolstov <v.tolstov@selfip.ru>

### CODE REVIEW

**Comment by rickard-von-essen** [comment]
Did some quick testing and it didn't work for me.

**Comment by vtolstov** [comment]
@rickard-von-essen thanks! I'm fix pr.

**Comment by vtolstov** [comment]
@rickard-von-essen i'm fix issue with go vet

File: common/iso_config.go [hunk 1]
```diff
@@ -125,14 +124,11 @@ func (c *ISOConfig) Prepare(ctx *interpolate.Context) ([]string, []error) {
 }
 
 func (c *ISOConfig) parseCheckSumFile(rd *bufio.Reader) error {
+	errNotFound := fmt.Errorf("No checksum for \"%s\" found at: %s", filepath.Base(c.ISOUrls[0]), c.ISOChecksumURL)
```
Comment (nightlyone) [inline common/iso_config.go]: You could use `%q` instead of `\"%s\"`

**Comment by vtolstov** [comment]
@cbednarski can you merge this?

**Comment by cbednarski** [comment]
Hmm hard to see what is happening without an example / test case. Can you add one?

It looks like the case you handled is an empty file. Did you intend to handle a file that has a checksum without a newline at the EOF? My reading of the docs suggests that this will complain if there is a checksum but the file does not end with `\n` but like I said, hard to tell without running the code.

**Comment by vtolstov** [comment]
i'm add test for newline

**Comment by cbednarski** [comment]
LGTM, thanks!

**Pull request merged**